src/UI/MainWindow.vala

```diff
@@ -428,6 +508,39 @@ public class Spreadsheet.UI.MainWindow : ApplicationWindow {
         expression.text = "";
 
         app_stack.set_visible_child_name ("welcome");
+
+        if (Spreadsheet.App.settings.get_strv ("recent-files").length != 0) {
+            if (recent_widgets_box == null) {
+                welcome_box.pack_start (create_recents_view ());
+            }
+
+            update_listview ();
+        }
+    }
+
+    private Gtk.Box create_recents_view () {
+        var title = new Gtk.Label (_("Recent files"));
+        title.halign = Gtk.Align.CENTER;
+        title.margin = 24;
+        title.get_style_context ().add_class (Granite.STYLE_CLASS_H2_LABEL);
+
+        var recent_files_scrolled = new Gtk.ScrolledWindow (null, null);
```

Comment:
Setting `halign` to `CENTER` for this widget makes the margins behave nicer when resizing the window IMO.

Reply:
Thank you for your investigation! I addressed.